user_manual.md

```diff
@@ -245,120 +289,157 @@ In particular, this means that any computation that pattern matches `d` will not
 Intuitively, this definition is introducing a new function, indexed by a function, that is of type `(-> Int Int)`.
 After parsing, the term `(@purify_fun f a)` is a function application whose operator is `(@purify_fun f)` and has a single child `a`.
 
-> Opaque arguments should always be expected before other arguments. Otherwise all applications of the given function will be ill-typed.
+> __Note:__ Opaque arguments should always be expected before other arguments. Otherwise all applications of the given function will be ill-typed.
+
+<a name="attributes"></a>
+
+### Declarations with attributes
+
+The Eunoia language supports term annotations on declared constants, which for instance allow the user to treat a constant as being variadic, i.e. taking an arbitrary number of arguments. The available annotations for this purpose are:
+
+- `:right-assoc` (resp. `:left-assoc`) denoting that application of the declared binary constant to more than two terms are to be treated as right (resp. left) associative,
 
-## <a name="attributes"></a>Declarations with attributes
+- `:right-assoc-nil <term>` (resp. `:left-assoc-nil <term>`) denoting that applications of the declared binary constant to one or more terms are to be treated as right (resp. left) associative, with the given `<term>` used as an additional rightmost (resp. leftmost) argument.
+<!--CT Do we really want to restrict the nil element to be a symbol, as opposed to a term? -->
+
+- `:chainable <symbol>` denoting that the arguments of the declared binary constant are chainable using the (binary) operator given by `<symbol>`,
 
-The Eunoia language supports term annotations on declared constants, which for instance can allow the user to treat a constant as being variadic, i.e. taking an arbitrary number of arguments. The available annotations in the Ethos for this purpose are:
-- `:right-assoc` (resp. `:left-assoc`) denoting that the declared constant is right (resp. left) associative,
-- `:right-assoc-nil <symbol>` (resp. `:left-assoc-nil <symbol>`) denoting that the declared constant is right (resp. left) associative with the given nil terminator given by `<symbol>`,
-- `:chainable <symbol>` denoting that the arguments of the declared constant are chainable using the (binary) operator given by `<symbol>`,
 - `:pairwise <symbol>` denoting that the arguments of the declared constant are treated pairwise using the (binary) operator given by `<symbol>`.
+
 - `:binder <symbol>` denoting that the first argument of the declared constant can be provided using a syntax for variable lists whose constructor is the one provided by `<symbol>`.
+
 A declared function can be marked with at most one of the above attributes or an error is thrown.
 
-A parameter may be marked with the following attributes:
+A parameter in a function symbol declaration may be marked with the following attribute:
```

Comment:
```suggestion
A parameter may be marked with the following attribute:
```

Reply:
Parameters may appear in function symbol declaration, but also in define, program, declare-rule.